src/components/organisms/EntityDetails/EntityDetailsContent/Settings/SettingsScheduling/Schedule.tsx

```diff
@@ -0,0 +1,189 @@
+import {useContext, useState} from 'react';
+
+import {Select, Tooltip} from 'antd';
+
+import {WarningOutlined} from '@ant-design/icons';
+
+import parser from 'cron-parser';
+import {formatDuration, intervalToDuration} from 'date-fns';
+
+import {Entity} from '@models/entity';
+
+import {Text} from '@custom-antd';
+
+import {ConfigurationCard, notificationCall} from '@molecules';
+
+import {displayDefaultErrorNotification, displayDefaultNotificationFlow} from '@utils/notification';
+import {uppercaseFirstSymbol} from '@utils/strings';
+
+import {useUpdateTestSuiteMutation} from '@services/testSuites';
+import {useUpdateTestMutation} from '@services/tests';
+
+import Colors from '@styles/Colors';
+import Fonts from '@styles/Fonts';
+
+import {EntityDetailsContext} from '@contexts';
+
+import {StyledSpace} from '../Settings.styled';
+import CronInput from './CronInput';
+import {StyledColumn, StyledCronFormat, StyledRow} from './Schedule.styled';
+
+const namingMap: {[key in Entity]: string} = {
+  'test-suites': 'test suite',
+  tests: 'test',
+};
+
+const quickOptions = [
+  {
+    label: 'Every 5 minutes',
+    value: '*/5 * * * *',
+  },
+  {
+    label: 'Every 30 minutes',
+    value: '*/30 * * * *',
+  },
+  {
+    label: 'Every hour',
+    value: '0 * * * *',
+  },
+  {
+    label: 'Every day',
+    value: '0 0 * * *',
+  },
+  {
+    label: 'Every week',
+    value: '0 0 * * 0',
+  },
+  {
+    label: 'Every month',
+    value: '0 0 1 * *',
+  },
+];
+
+const defaultCronString = '* * * * *';
+
+const Schedule: React.FC = () => {
+  const {entity, entityDetails} = useContext(EntityDetailsContext);
+
+  const [updateTest] = useUpdateTestMutation();
+  const [updateTestSuite] = useUpdateTestSuiteMutation();
+
+  const updateRequestsMap: {[key in Entity]: any} = {
+    'test-suites': updateTestSuite,
+    tests: updateTest,
+  };
+  const name = entityDetails?.name;
+  const schedule = entityDetails?.schedule;
+
+  const [cronString, setCronString] = useState(schedule || defaultCronString);
+  const [wasTouched, setWasTouched] = useState(false);
+  const [isValidFormat, setIsValidFormat] = useState(true);
+
+  if (!entity || !entityDetails) {
+    return null;
+  }
+
+  const onSave = () => {
+    updateRequestsMap[entity]({
+      id: name,
+      data: {
+        ...entityDetails,
+        schedule: cronString,
+      },
+    })
+      .then((res: any) => {
+        displayDefaultNotificationFlow(res, () => {
+          notificationCall('passed', `${uppercaseFirstSymbol(namingMap[entity])} schedule was successfully updated.`);
+        });
+        setWasTouched(false);
```

Comment:
It's also ran on error (so the form can't be sent once again without changing it) - but if it's similar for other settings, I believe that it can be ignored for now, and fixed later globally

Reply:
Yes, we actually have to refactor this for all settings forms